beacon-chain/node/node.go

```diff
@@ -296,17 +297,26 @@ func (b *BeaconNode) startStateGen() {
 	b.stateGen = stategen.New(b.db, b.stateSummaryCache)
 }
 
+func readbootNodes(fileName string) ([]string, error) {
+	fileContent, err := ioutil.ReadFile(fileName)
+	if err != nil {
+		return nil, err
+	}
+	listNodes := make([]string, 0)
+	yaml.Unmarshal(fileContent, &listNodes)
+	return listNodes, nil
+}
+
 func (b *BeaconNode) registerP2P(cliCtx *cli.Context) error {
-	// Bootnode ENR may be a filepath to an ENR file.
-	bootnodeAddrs := strings.Split(cliCtx.String(cmd.BootstrapNode.Name), ",")
-	for i, addr := range bootnodeAddrs {
-		if filepath.Ext(addr) == ".enr" {
-			b, err := ioutil.ReadFile(addr)
-			if err != nil {
-				return err
-			}
-			bootnodeAddrs[i] = string(b)
+	bootnodeAddrs := make([]string, 0)
+	var err error
+	if bootstrapfileName := cliCtx.String(cmd.BootStrapNodeFile.Name); bootstrapfileName != "" {
+		bootnodeAddrs, err = readbootNodes(bootstrapfileName)
```

Comment:
If I omitted this (line 312) and changed = to := on line 314, then it would consider the inner bootnodeAddrs a new variable (shadowing I guess).